### PR TITLE
Pie

### PR DESCRIPTION
**Pull request by raysenlau**
add nokia B2N and DRG:
. Lift hand On
. Double tap wake
. Comfortable darkness backlight
. Nokia 6.1 Plus perfect notch

add lenovo Z5:
. notch

### CODE REVIEW

File: Lenovo/Z5/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+        package="me.phh.treble.overlay.lenovo.z5"
+        android:versionCode="1"
+        android:versionName="1.0">
+        <overlay android:targetPackage="android"
+                android:requiredSystemPropertyName="ro.vendor.product.name"
```
Comment (phhusson) [inline Lenovo/Z5/AndroidManifest.xml]: Please match on vendor fingerprint

File: Nokia/b2n/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+        package="me.phh.treble.overlay.nokia.b2n"
+        android:versionCode="1"
+        android:versionName="1.0">
+        <overlay android:targetPackage="android"
+                android:requiredSystemPropertyName="ro.vendor.build.fingerprint"
+                android:requiredSystemPropertyValue="+*Onyx*"
```
Comment (phhusson) [inline Nokia/b2n/AndroidManifest.xml]: Please also match with the manufacturer
+nokia/Onyx* or something like that

File: Nokia/drg/AndroidManifest.xml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+        package="me.phh.treble.overlay.nokia.drg"
+        android:versionCode="1"
+        android:versionName="1.0">
+        <overlay android:targetPackage="android"
+                android:requiredSystemPropertyName="ro.vendor.build.fingerprint"
+                android:requiredSystemPropertyValue="+*Dragon*"
```
Comment (phhusson) [inline Nokia/drg/AndroidManifest.xml]: Please also match with the manufacturer

File: Nokia/drg/res/values/config.xml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+	<integer-array name="config_virtualKeyVibePattern">
+        <item>50</item>
+    </integer-array>
+
+	<integer-array name="config_fingerprintSuccessPattern">
+        <item>0</item>
+        <item>75</item>
+    </integer-array>
+    <integer-array name="config_fingerprintFailPattern">
```
Comment (phhusson) [inline Nokia/drg/res/values/config.xml]: This doesn't exist in AOSP...? (same for both fail and success)

File: Nokia/drg/res/values/config.xml [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        <item>0</item>
+        <item>75</item>
+        <item>100</item>
+        <item>75</item>
+    </integer-array>
+
+    <bool name="config_enableCpuBoostForTouch">true</bool>
```
Comment (phhusson) [inline Nokia/drg/res/values/config.xml]: Doesn't exist in AOSP.
If other ROMs uses it, please point it to me, I can accept to merge than then

File: Nokia/drg/res/values/config.xml [hunk 6]
```diff
@@ -0,0 +1,117 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+	<bool name="config_swipe_up_gesture_setting_available">true</bool>
```
Comment (phhusson) [inline Nokia/drg/res/values/config.xml]: Well, IMO that's not an hardware decision, but a ROM decision.
But whatever.

File: Nokia/drg/res/values/config.xml [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+<resources>
+	<bool name="config_swipe_up_gesture_setting_available">true</bool>
+    <bool name="config_supportSystemNavigationKeys">true</bool>
+    <bool name="config_smart_battery_available">true</bool>
+    <bool name="config_supportDoubleTapWake">true</bool>
+    <bool name="config_showNavigationBar">true</bool>
+    <bool name="config_enableMultiUserUI">true</bool>
+    <bool name="config_camera_sound_forced">false</bool>
+    <bool name="config_dozePulsePickup">true</bool>
+    <bool name="config_display_rat">false</bool>
```
Comment (phhusson) [inline Nokia/drg/res/values/config.xml]: It doesn't exist in AOSP

File: Nokia/drg/res/values/config.xml [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+    <bool name="config_useRoundIcon">true</bool>
```
Comment (phhusson) [inline Nokia/drg/res/values/config.xml]: That's a ROM decision, not hardware decision.
But anyway, this value is obsolete and used by noone.

File: Nokia/drg/AndroidManifest.xml [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+				android:priority="15"
```
Comment (phhusson) [inline Nokia/drg/AndroidManifest.xml]: Please use not already selected priorities.
You can take 47 48 49.

**Comment by phhusson** [comment]
Please PR on master rather than pie
(don't worry I'll rebase pie on top of master)

**Comment by raysenlau** [comment]
Sir, check it please.

**Comment by phhusson** [comment]
Merged in https://github.com/phhusson/vendor_hardware_overlay/commit/7535d704873b969685acf77040f00590caaf1e7b

Thanks

**Pull request closed without merging**